lib/medtronic600/medtronic600Simulator.js

```diff
@@ -510,13 +382,122 @@ class Medtronic600Simulator {
     });
   }
 
+  applySuspendedBasals() {
```

Comment:
This function is really hard to thoroughly review by just looking at the code. Do we have sufficient test strategies to make sure all of this works as expected?